src/Microsoft.ML.OnnxTransform/OnnxUtils.cs

```diff
@@ -73,10 +73,44 @@ public OnnxNodeInfo(string name, OnnxShape shape, System.Type type)
         public readonly List<string> InputNames;
         public readonly List<string> OutputNames;
 
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
         public OnnxModel(string modelFile)
+            : this (modelFile, -1, false)
+        {
+        }
+
+        /// <summary>
+        /// Constructs OnnxModel object from file.
+        /// </summary>
+        /// <param name="modelFile">Model file path</param>
+        /// <param name="gpuDeviceId">Gpu device ID to execute on. -1 for CPU</param>
+        /// <param name="fallbackToCpu">If true, resumes CPU execution quitely upon GPU error</param>
+        public OnnxModel(string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false)
         {
             _modelFile = modelFile;
```

Comment:
What about checking the file path? Should this throw a FileNotFoundException if the file is not found?

Reply:
We don't need a separate file check. Onnxruntime checks for this already.